tests/precommit_test.py

```diff
@@ -131,12 +138,56 @@ def test_get_added_and_existing_findings_in_changes_for_changes(self):
         self.assert_findings_ids(self.precommit_client.findings_in_changed_code, [])
         self.assert_findings_ids(self.precommit_client.existing_findings, [4, 5, 6, 7])
 
+    @responses.activate
+    def test_only_upload_files_in_project_subpath(self):
+        project_subpath = 'project'
+        path_outside_project = 'another_path'
+
+        changed_file_in_project = os.path.join(project_subpath, ANALYZED_FILE_NAME)
+        changed_file_not_in_project = os.path.join(path_outside_project, ANALYZED_FILE_NAME)
+        deleted_file_in_project = os.path.join(project_subpath, DELETED_FILE_NAME)
+        deleted_file_not_in_project = os.path.join(path_outside_project, DELETED_FILE_NAME)
+
+        self.precommit_client = self._get_precommit_client(
+            {changed_file_in_project: '',
+             changed_file_not_in_project: ''},
+            [deleted_file_in_project, deleted_file_not_in_project],
+            project_subpath=project_subpath)
+        self.mock_precommit_findings_churn()
+
+        self.precommit_client.run()
+
+        precommit_request = next(call.request for call in responses.calls if call.request.method == 'PUT')
+
+        self.assertIn(changed_file_in_project, precommit_request.body)
+        self.assertNotIn(changed_file_not_in_project, precommit_request.body)
+        self.assertIn(deleted_file_in_project, precommit_request.body)
+        self.assertNotIn(deleted_file_not_in_project, precommit_request.body)
+
+    @responses.activate
+    def test_only_print_findings_in_project_subpath(self):
+        project_subpath = 'project'
+
+        self.precommit_client = self._get_precommit_client(self._get_no_changed_files(), self._get_no_deleted_files(),
+                                                           project_subpath=project_subpath, fetch_all_findings=True)
+        # The mocked finding does not have the project sub-path
+        self.mock_existing_findings(CURRENT_BRANCH, [1])
```

Comment:
I'm a bit confused by the setup... how can there be a findings, if there are no changed files?

Reply:
Note the `fetch_all_findings=True` which fetches all findings in Teamscale and not only the ones for files relevant for pre-commit.

Reply:
😆